packages/core/useResize/index.ts

```diff
@@ -0,0 +1,336 @@
+    return unref(edges).includes(edge)
+  }
+
+  function handlePointer({ currentX, currentY }: typeof pointer) {
+    if (isResizing.value || isOutside.value)
+      return
+
+    let { left, right, top, bottom } = target.value!.getBoundingClientRect()
+
+    left += 1
+    right -= 1
+    top += 1
+    bottom -= 1
+
+    if (
+      ((currentY - top) < 2 || (currentX - left) < 2)
+      && Math.abs(currentY - top) < 8
+      && Math.abs(currentX - left) < 8
+      && isEdgeActive('top-left')
+      && isOnForeground(left, top, true, 1, 1)
+    )
+      setCursorAndDirection('nwse-resize', 'top-left')
+
+    else if (
+      ((currentY - top) < 2 || (currentX - right) > -2)
+      && Math.abs(currentY - top) < 8
+      && Math.abs(currentX - right) < 8
+      && isEdgeActive('top-right')
+      && isOnForeground(right, top, true, -1, 1)
+    )
+      setCursorAndDirection('nesw-resize', 'top-right')
+
+    else if (
+      ((currentY - bottom) > -2 || (currentX - left) < 2)
+      && Math.abs(currentY - bottom) < 8
+      && Math.abs(currentX - left) < 8
+      && isEdgeActive('bottom-left')
+      && isOnForeground(left, bottom, true, 1, -1)
+    )
+      setCursorAndDirection('nesw-resize', 'bottom-left')
+
+    else if (
+      ((currentY - bottom) > -2 || (currentX - right) > -2)
+      && Math.abs(currentY - bottom) < 8
+      && Math.abs(currentX - right) < 8
+      && isEdgeActive('bottom-right')
+      && isOnForeground(right, bottom, true, -1, -1)
+    )
+      setCursorAndDirection('nwse-resize', 'bottom-right')
+
+    else if (
+      (currentY - bottom) < 8
+      && (currentY - bottom) >= 0
+      && currentX > left
+      && currentX < right
+      && isEdgeActive('bottom')
+      && isOnForeground(currentX, bottom)
+    )
+      setCursorAndDirection('ns-resize', 'bottom')
+
+    else if (
+      (currentY - top) > -8
+      && (currentY - top) <= 0
+      && currentX > left
+      && currentX < right
+      && isEdgeActive('top')
+      && isOnForeground(currentX, top)
+    )
+      setCursorAndDirection('ns-resize', 'top')
+
+    else if (
+      (currentX - left) > -8
+      && (currentX - left) <= 0
+      && currentY > top
+      && currentY < bottom
+      && isEdgeActive('left')
+      && isOnForeground(left, currentY)
+    )
+      setCursorAndDirection('ew-resize', 'left')
+
+    else if (
+      (currentX - right) < 8
+      && (currentX - right) >= 0
+      && currentY > top
+      && currentY < bottom
+      && isEdgeActive('right')
+      && isOnForeground(right, currentY)
+    )
+      setCursorAndDirection('ew-resize', 'right')
+
+    else
+      setCursorAndDirection('', '', '')
+  }
+
+  function setSize(width: number, height: number) {
+    target.value!.style.width = `${clamp(width, unref(minWidth) as number, unref(maxWidth) as number)}px`
+    target.value!.style.height = `${clamp(height, unref(minHeight) as number, unref(maxHeight) as number)}px`
+  }
+
+  function useElementSize() {
```

Comment:
This can probably be moved to the top level as well so we don't have to manually clean it up